### ISSUE TITLE
Build script to check and report of any Stylelint issues

### ISSUE THREAD

**Issue by coliff** [h5bp/main.css#34]
**I'm submitting a ...**

- [ ] bug report
- [x] feature request
- [ ] other (Please do not submit support requests here (below))

It'd be good if the Gulp build script automatically checked and either fix or report/warn of any Stylelint issues.

**Comment by roblarsen** [comment]
👍 

We could alternatively add with [pre\-commit \- npm](https://www.npmjs.com/package/pre-commit) as a pre-commit hook running an npm task.

**Comment by roblarsen** [comment]
(related thought: pre-commit might be useful to add to all the projects that are generated. Write a test to ensure that the dist folder has been updated, run that when code is committed)

**Comment by coliff** [comment]
yep, that sounds like a great idea.

**Comment by coliff** [comment]
I started looking into pre-commit hooks a few days ago and got it working well with [husky](https://github.com/typicode/husky) and [lint-staged](https://github.com/okonet/lint-staged). I can make a PR so you can take a look/test. It's essentially `package.json` with this:

```
{
  "name": "main.css",
  "version": "2.0.0",
  "description": "The main.css file from HTML5 Boilerplate",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/h5bp/main.css.git"
  },
  "keywords": [
    "css",
    "h5bp"
  ],
  "author": "@h5bp",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/h5bp/main.css/issues"
  },
  "homepage": "https://github.com/h5bp/main.css#readme",
  "devDependencies": {
    "@babel/register": "^7.0.0",
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.3",
    "babel-preset-env": "^1.7.0",
    "del": "^4.1.1",
    "gulp": "^3.9.1",
    "gulp-autoprefixer": "^6.1.0",
    "gulp-cssimport": "^7.0.0",
    "gulp-header": "^2.0.7",
    "gulp-load-plugins": "^1.6.0",
    "husky": "^2.4.1",
    "lint-staged": "^8.2.1",
    "prettier": "^1.18.2",
    "stylelint": "^10.1.0",
    "stylelint-config-recommended": "^2.2.0"
  },
  "babel": {
    "presets": [
      [
        "env",
        {
          "targets": {
            "node": "current"
          }
        }
      ]
    ]
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged"
    }
  },
  "lint-staged": {
    "linters": {
      "*.{css,yml,md}": [
        "prettier --write",
        "git add"
      ],
      "*.{css}": [
        "stylelint --fix",
        "git add"
      ]
    },
    "ignore": [
      "README.md"
    ]
  }
}
```

**Comment by coliff** [comment]
Todo: add eslint like this: https://github.com/lipis/prettier-setup

**Comment by roblarsen** [comment]
@coliff want to revisit this?

**Comment by coliff** [comment]
Yep! I'll open a PR tomorrow for this.

**Comment by roblarsen** [comment]
closed via #55
